src/libos/src/util/sgx/sgx_attestation_agent.rs

```diff
@@ -133,8 +139,21 @@ impl InnerAgent {
         let report = rsgx_create_report(&self.target_info, report_data)
             .map_err(|_e| errno!(EINVAL, "sgx_error"))?;
         let mut qe_report = sgx_report_t::default();
-        // TODO: what if quote_buf is not big enough?
-        let mut quote_buf = [0_u8; 4096];
+        let mut quote_len: u32 = 0;
+        let mut rt = Default::default();
+        let status = unsafe {
+            occlum_ocall_sgx_calc_quote_size(&mut rt as _, sigrl_ptr, sigrl_size, &mut quote_len as _)
+        };
+        assert!(status == sgx_status_t::SGX_SUCCESS);
+        if rt != sgx_status_t::SGX_SUCCESS {
+            match rt {
+                sgx_status_t::SGX_ERROR_BUSY => {
```

Comment:
According to Intel SGX SDK's manual, the only possible error is `SGX_ERROR_INVALID_PARAMETER`. So no need to check `SGX_ERROR_BUSY`.

Reply:
Right,  fixed.